kernel/userspace.c

```diff
@@ -683,23 +707,23 @@ char *z_user_string_alloc_copy(char *src, size_t maxlen)
 
 int z_user_string_copy(char *dst, char *src, size_t maxlen)
 {
-	unsigned long actual_len;
```

Comment:
This was deliberately made an unsigned long due to the way size_t is defined by some C libraries.
Can you please confirm that there are no issues when building with newlib?